uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/LeaveRequestCalendarFeedConfig.php

```diff
@@ -232,4 +232,185 @@ public static function getValuesArray($id) {
       return [];
     }
   }
+
+  /**
+   * Implementation of the addSelectWhereClause:
+   * This ensures that the administrator (user with 'can administer calendar feeds')
+   * permission can access all feed configs while users without this permission can
+   * only access feed configs which they have visibility access to. i.e if they belong
+   * to any of the department or location visibility set  for a feed config.
+   */
+  public function addSelectWhereClause() {
+    if (CRM_Core_Permission::check([['can administer calendar feeds']])) {
+      return;
+    }
+
+    $limitedVisibilityFeedConfigs = $this->getLimitedVisibilityFeedConfigs();
+
+    if (empty($limitedVisibilityFeedConfigs)) {
+      return;
+    }
+
+    $accessibleDepartments = $this->getAccessibleDepartmentsForLoggedInContact();
+    $accessibleLocations = $this->getAccessibleLocationsForLoggedInContact();
+
+    $inaccessibleFeedConfigs = $this->getNotAccessibleFeedConfigs(
+      $limitedVisibilityFeedConfigs,
+      $accessibleDepartments,
+      $accessibleLocations
+    );
+
+    if (empty($inaccessibleFeedConfigs)) {
+      return;
+    }
+
+    $clauses['id'] = 'NOT IN (' . implode(',', $inaccessibleFeedConfigs) . ')';
+
+    CRM_Utils_Hook::selectWhereClause($this, $clauses);
```

Comment:
Just curious, why are you calling the hook here? Is this done in the parent? If it is then is it an option to do a `parent::addSelectWhereClause` here instead?

Reply:
I tried adding parent::addSelectWhereClause but the tests were failing even though the parent calls the hook. I could take a closer look later, although we have this same implementation for the LeaveRequest BAO.